File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.Specification;
import com.pinyougou.pojo.TbSpecification;

import java.util.List;
import java.util.Map;

public interface SpecificationService {
    /** 
    * @Description: 查找所有 
    * @Param: [] 
    * @return: java.util.List<com.pinyougou.pojo.TbSpecification> 
    * @Author: Yin 
    * @Date: 2018/9/2 
    */ 
    List<TbSpecification> findAll();

    /** 
    * @Description: 根据条件查询 
    * @Param: [page, rows, specification] 
    * @return: java.util.Map<java.lang.String,java.lang.Object> 
    * @Author: Yin 
    * @Date: 2018/9/2 
    */ 
    Map<String,Object> findByCondition(int page, int rows, TbSpecification specification);

    /** 
    * @Description: 添加规格
    * @Param: [specification] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/2
     * @param specification
    */ 
    void addSpecification(Specification specification);

    /** 
    * @Description: 修改实体的信息
    * @Param: [specification] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/2 
    */ 
    void updateSpecification(Specification specification);

    /** 
    * @Description: 根据id去寻找规格的信息 
    * @Param: [id] 
    * @return: com.pinyougou.entity.Specification 
    * @Author: Yin 
    * @Date: 2018/9/2 
    */ 
    Specification findByOne(Long id);

    /** 
    * @Description: 批量删除 
    * @Param: [ids] 
    * @return: void 
    * @Author: Yin 
    * @Date: 2018/9/2
     * @param ids
    */ 
    void deleteSpecification(Long[] ids);
}

File: pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
//服务层
app.service('specificationService',function ($http) {
    this.findAll=function () {
        return $http.get("../specification/findAll.do");
    }
    /*条件查询*/
    this.findByCondition=function (page,rows,entity) {
        return $http.post("../specification/findByCondition.do?page="+page+"&rows="+rows,entity)
    }

    /*增加规格*/
    this.addSpecification=function (entity) {
        var methodName = "add";
        if(entity.specification.id!=null){
            //这个时候应该进行的是修改操作
            methodName = "update";
        }
        return $http.post("../specification/"+methodName+".do",entity);
    }
    /*查询*/
    this.findOne=function (id) {
        return $http.get("../specification/findOne.do?id="+id);
    }

    /*删除的操作*/
    this.dele=function (ids) {
        return $http.get("../specification/delete.do?ids="+ids);
    }

    /*供应给模板的下拉列表的操作*/
    this.selectOptionSpec=function () {
        return $http.get("../specification/selectOptionSpec.do");
    }

    /*增加的操作*/
    this.add=function (entity) {
        return $http.post("../specification/add.do",entity);
    }

})